independent-projects/tools/codestarts/codestarts.adoc

```diff
@@ -0,0 +1,220 @@
+----
+
+* other files are just copied as is.
+
+
+=== Writing Example Code for an Extension
+
+Let's imagine we are writing the example codestart for my "foo" extension. As we discussed before, the base of the project is already provided so we can focus on the example code.
+
+1. Create your `codestart.yml`:
+
+codestart.yml:
+[source,yaml]
+----
+---
+name: foo-example
+ref: foo
+type: example
+missing-languages:
+  - kotlin
+  - scala
+spec:
+  base:
+    data:
+      # Some data for my templates which gives could be overridden by the user input
+      some:
+        data:
+          here: bar
+          there: foo
+    dependencies:
+      # the dependencies needed for my codestart (note that the extension dependency is auto-added if missing)
+      - io.quarkus:quarkus-foo
+      - io.quarkus:quarkus-resteasy
+      # ... you can also provide a version (when it's not part of the platform bom)
+      - group:artifact:version
+    test-dependencies:
+      # test dependencies
+      - io.rest-assured:rest-assured
+----
+
+*NOTE* You don't need to add `pom.xml` or `build.gradle`, it is going to be auto generated. It will include the dependencies provided in the `codestart.yml`.
+
+2. Add you example code in java, kotlin or scala
+
+* java/src/main/java/org/acme/foo/Foo.java:
+[source,java]
+----
+package org.acme.foo
+
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.core.MediaType
+
+@Path("foo")
+class ExampleResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    fun hello() = "Hello foo"
+}
+----
+
+* kotlin/src/main/kotlin/org/acme/foo/Foo.kt
+* scala/src/main/scala/org/acme/foo/Foo.kt
+
+*NOTE* Just add `.tpl.qute` if you want it to be rendered with qute and use some data
+
+*NOTE* Most of the time you won't need it for code, but you can use a `base` directory to add files to process for all languages.
+
+
+2. Add some configuration (if needed)
+
+The `base/src/main/resources/application.yml` file is going to be merged with the other codestarts config and automatically converted to the selected config type (yaml or properties).
+
+Note: if it's specific to a language, you can put the config in each languages folders
+
+* base/src/main/resources/application.yml (using quarkus yaml config convention):
+[source,yaml]
+----
+quarkus:
+  http:
+    port: 8081
+----
+
+3. Create a readme section (if needed)
+
+This will be appended to all different selected examples.
+
+* base/README.md
+[source,markdown]
+----
+# FOO Example
+
+The Foo Example shows...
+----
+
+=== Tips for writing extension example code
+
+- You don't need to care about buildtool, dockerfiles, ...
+- Use the package `org.acme.[unique-name]` for your sources.
+- Use the path `/unique-name/...` for your REST paths
+- Write tests for your examples
+- Examples in different codestarts must be independent, only the config and the build file are merged.
```

Comment:
move this up as the first - repeat it at the end :)

Reply:
Sure I'll make it red also

Reply:
add a blink tag :) (j/k)